### ISSUE TITLE
CA1001 not emitted for IDisposable fields

### ISSUE THREAD

**Issue by pharring** [dotnet/roslyn-analyzers#1562]
#### Analyzer package

Microsoft.CodeQuality.CSharp.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.CSharp.Analyzers/2.6.0)

#### Analyzer

CA1001: Types that own disposable fields should be disposable

#### Repro steps

Analyze this:
```C#
using System;
using System.IO;

namespace ClassLibrary1
{
    public class Class1
    {
        private readonly Stream _disp1 = new MemoryStream();
    }

    public class Class2
    {
        private readonly IDisposable _disp1 = new MemoryStream();
    }
}

```

#### Expected behavior

CA1001 should be reported on both Class1 and Class2

#### Actual behavior

CA1001 on Class1 but _not_ on Class2

**Comment by sharwell** [comment]
:+1:
